accesscontrol/jwt.go

```diff
@@ -167,7 +196,24 @@ func (j *JWT) Validate(req *http.Request) error {
 	return nil
 }
 
-func (j *JWT) getValidationKey(_ *jwt.Token) (interface{}, error) {
+func (j *JWT) getValidationKey(token *jwt.Token) (interface{}, error) {
+	if j.jwks != nil {
+		id := token.Header["kid"]
+		algorithm := token.Header["alg"]
+		if id == nil || algorithm == nil {
```

Comment:
split these tests?

Reply:
If `kid` is optional in JWKS, do we have to require the `kid` header here?

Reply:
At least: `if id == nil || id == "" || …`